pkg/controller/plan/validation.go

```diff
@@ -736,6 +741,38 @@ func (r *Reconciler) validateHooks(plan *api.Plan) (err error) {
 }
 
 func (r *Reconciler) validateVddkImage(plan *api.Plan) (err error) {
+	source := plan.Referenced.Provider.Source
+	if source == nil {
+		return
```

Comment:
We want to return nil only if everything passes, would it make sense to return error here?
I know it is just an edge case

Reply:
+ think that this should be already coverd in previous validation but don't mind having extra protection